ca_pool_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package identity
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestID_CaPoolGetChainDirectChild(t *testing.T) {
+	idc := Config{
+		Key:  "./testdata/root/keys/client1.key",
+		Cert: "./testdata/root/certs/client1.cert",
+		CA:   "./testdata/chain.pem",
+	}
```

Comment:
Hard to tell if these test files are correct. I'd prefer these to be built-in code as fresh fixtures like `chains_test.go` does it. It is longer, but it is easy to update and verify the correctness.

I won't stop the PR for it, but something for the future.

Reply:
I'll see if I can do that when I add more validation